ntbs-service/Models/Notification.cs

```diff
@@ -1,18 +1,127 @@
         public virtual Episode Episode { get; set; }
         public virtual PatientTBHistory PatientTBHistory { get; set; }
         public virtual ContactTracing ContactTracing { get; set; }
         public virtual SocialRiskFactors SocialRiskFactors { get; set; }
+
+        public string FullName {
+            get {
+                return String.Join(", ", new string[] {PatientDetails.FamilyName.ToUpper(), PatientDetails.GivenName});
+            }
+        }
+
+        public string FormattedNhsNumber {
+            get {
+                return CreateFormattedNhsNumberString();
+            }
+        }
+
+        public string FormattedPostcode {
+            get {
+                return CreatePostcodeString();
+            }
+        }
+
+        public string SitesOfDiseaseList { 
+            get {
+                return CreateSitesOfDiseaseString();
+            } 
+        }
+
+        public string DrugRiskFactorTimePeriods { 
+            get {
+                return CreateTimePeriodsString(SocialRiskFactors.RiskFactorDrugs);
+            }
+        }
+        public string HomelessRiskFactorTimePeriods { 
+            get {
+                return CreateTimePeriodsString(SocialRiskFactors.RiskFactorDrugs);
+            }
+        }
+        public string ImprisonmentRiskFactorTimePeriods { 
+            get {
+                return CreateTimePeriodsString(SocialRiskFactors.RiskFactorDrugs);
+            }
+        }
+
+        public int? DaysFromOnsetToTreatment {
+            get {
+                return CalculateDaysBetweenNullableDates(ClinicalDetails.TreatmentStartDate, ClinicalDetails.SymptomStartDate);
+            }
+        }
+
+        public int? DaysFromOnsetToPresentation {
+            get {
+                return CalculateDaysBetweenNullableDates(ClinicalDetails.PresentationDate, ClinicalDetails.SymptomStartDate);
+            }
+        }
+
+        public int? DaysFromPresentationToDiagnosis {
+            get {
+                return CalculateDaysBetweenNullableDates(ClinicalDetails.DiagnosisDate, ClinicalDetails.PresentationDate);
+            }
+        }
+
+        public int? DaysFromDiagnosisToTreatment {
+            get {
+                return CalculateDaysBetweenNullableDates(ClinicalDetails.TreatmentStartDate, ClinicalDetails.DiagnosisDate);
+            }
+        }
+
+        public string BCGVaccinationStateAndYear {
+            get {
+                return FormatVaccintationStateAndYear(ClinicalDetails.BCGVaccinationState, ClinicalDetails.BCGVaccinationYear);
+            }
+        }
+
+        public string FormatVaccintationStateAndYear(Status vaccinationState, int? vaccinationYear) {
+            return vaccinationState.ToString() + (vaccinationYear != null ? " - " + vaccinationYear : "");
+        }
+
+        public int? CalculateDaysBetweenNullableDates(DateTime? date1, DateTime? date2) {
+            return (date1?.Date - date2?.Date)?.Days;
```

Comment:
Not sure this works correctly..
If `date1` is null, then `date?.Date` is also null, at which point we'll be trying to subtract from null - causing an exception I imagine 🤔

Reply:
I've tested this and it seems to just return null if either date is missing and doesn't throw an exception, happy to check for nulls before running this to make it safer though?

Reply:
No, you're right, looks like c# is safer than i thought! :)